File: me_daily/2025/11/27/FurnitureStore/FurnitureStore.cpp
#include <bits/stdc++.h>
#define For(i,n) for(int i = 0; i < (n);i++)
using namespace std;
using vi = vector<int>;

signed main() {
    int t;cin>>t;while(t--){
        int n;cin>>n;vi a(n);For(i,n)cin>>a[i];
        vi never_bought;
        int k = 0;
        int curr_min = 101;
        For(i, n) {
            if (a[i] >= curr_min) {
                k++;
                never_bought.push_back(i+1);
            }else {
                curr_min = min(curr_min, a[i]);
            }
        }
        cout<<k<<"\n";
        if (!never_bought.empty()) {
            For(i,never_bought.size()) {
            cout << never_bought[i] <<" ";
        }
        }
        
        cout <<"\n";
    }
    return 0;
}

File: me_daily/2025/11/27/FurnitureStore/FurnitureStore.py
from collections import defaultdict
from typing import List
import sys
from collections import Counter
input = lambda: sys.stdin.readline().strip()
II = lambda: int(input())
LII = lambda: list(map(int, input().split()))
SI = lambda: input()
LSI = lambda: input().split()
t = II()

for _ in range(t):
    n = II()
    a = LII()
    k = 0
    never_bought = []
    curr_min = float('inf')
    for i in range(n):
        if a[i] >= curr_min:
            k += 1
            never_bought.append(i+1)
        else:
            curr_min = min(a[i], curr_min)
    
    print(k)
    print(*never_bought)